lib/ansible/modules/package_facts.py

```diff
@@ -389,14 +388,12 @@ def list_installed(self):
         return out.splitlines()
 
     def get_package_details(self, package):
-        raw_pkg_details = {}
-        for line in package.splitlines():
-            m = re.match(r"([\w ].*?)-([0-9-\.]+[0-9a-z-\.]*-r[0-9]+)", to_native(line))
-            if m:
-                raw_pkg_details['name'] = m.group(1)
-                raw_pkg_details['version'] = m.group(2)
-
-        return raw_pkg_details
+        nvr = package.rsplit('-', 2)
+        return {
+            'name': nvr[0],
+            'version': nvr[1],
+            'release': nvr[2],
+        }
```

Comment:
This worked well in my testing. It can be made a bit more resilient in the event a package doesn't split as expected.
```suggestion
        raw_pkg_details = {'name', package}
        # or maybe
        # raw_pkg_details = {'name': package, 'version': '', 'release': ''}
        nvr = package.rsplit('-', 2)
        try:
            return {
                'name': nvr[0],
                'version': nvr[1],
                'release': nvr[2],
            }
        except IndexError:
            return raw_pkg_details
```